src/writer.jl

```diff
@@ -22,17 +22,27 @@ function write(path::AbstractString, obj; force=false)
         end
     end
 
-    # Handle tabular data
     if Tables.istable(obj)
-        geomcol = first(GI.geometrycolumns(obj))
-        geoms = Tables.getcolumn(obj, geomcol)
-        @warn "DBFTables.jl does not yet `write`, so only .shp, .shx, and .prj files can be written."
-        # DBFTables.Table(obj) # TODO remove geom column
-        # DBFTables.write # TODO DBF write function
+        sch = Tables.schema(obj)
+        tbl = Dict(field => Tables.getcolumn(obj, field) for field in sch.names)
+        # find first column that only contains geometries (and possibly missing)
+        geomcol = findfirst(tbl) do data
+            all(x -> ismissing(x) || GI.geomtrait(x) isa GI.AbstractGeometryTrait, data) && any(!ismissing, data)
+        end
+        geoms = pop!(tbl, geomcol)
     elseif GI.geomtrait(obj) isa GI.AbstractGeometryCollectionTrait
         geoms = (GI.getgeom(obj, i) for i in 1:GI.ngeom(obj))
+        tbl = fill((;feature=missing), length(geoms))  # fill .dbf with missing so that it exists
+    elseif GI.geomtrait(obj) isa GI.AbstractFeatureCollectionTrait
+        tbl = GI.properties.(obj)
+        geoms = GI.geometry.(obj)
+        # geomcol = first(GI.geometrycolumns(obj))
+        # geoms = GI.getgeom(obj)
+        # features = GI.getfeature(obj)
+        # tbl = features
     else
         geoms = obj
+        tbl = fill((;feature=missing), length(geoms))
```

Comment:
Is the idea that there would be an empty `feature` column if there is no data for the table?

Can we not just have a totally empty table?

Reply:
> The dBASE table contains feature attributes with one record per feature. The one-to-one relationship between geometry and attributes is based on record number. Attribute records in the dBASE file must be in the same order as records in the main file.

- ESRI Shapefile Technical Description: https://www.esri.com/content/dam/esrisites/sitecore-archive/Files/Pdfs/library/whitepapers/pdfs/shapefile.pdf

Reply:
Ok makes sense

Reply:
But we could just *not* write the file in case of an empty table right?

Reply:
Yeah that's another option. `Table` doesn't wirk without a .dbf present. It should at least round trip properly so we could fix the `isfile(paths.dbf)` check instead to give `@info` rather than an error, and let the dbf field be `nothing`.

Reply:
Reading the spec it seems we must write a table, so this should work.